dig/default.go

```diff
@@ -20,30 +20,30 @@
 
 package dig
 
-// Default graph is used for all the top-level inject calls
-var defaultGraph = newGraph()
+// DefaultGraph is used for all the top-level inject calls
+var DefaultGraph Graph = newGraph()
```

Comment:
I was hoping to make something like:
```golang
const _middlewareKey = "_httpClientMiddleware"
const _graphKey = "_httpClientGraph"

func WithOutbound(middleware... OutboundMiddleware) modules.Option{
	return func(info *service.ModuleCreateInfo) error {
		info.Items[_middlewareKey] = append(info.Items[_middlewareKey].([]OutboundMiddleware, middleware...)
		return nil
	}
}

func WithGraph(graph dig.Graph) modules.Option {
	return func(info *service.ModuleCreateInfo) error {
		info.Items[_graphKey] = graph
		return nil
	}
}

func New(options... modules.Option) *http.Client {
	var info service.ModuleCreateInfo
	for _, opt := range options {
		opt(info)
	}

	graph := dig.DefaultGraph()
	if g, ok := info.Items[_graphKey]; ok {
		graph = g.(dig.Graph)
	}

	middleware := make([]OutboundMiddleware, 0, len(options) + 2)

	var tracer opentracing.Tracer
	if err := graph.Resolve(&tracer); err != nil {

	}

	middleware = append(middleware, tracingOutbound(*tracer), authenticationOutbound(info))
}
```

This way a default client will be created without any extra parameters that users need to pass, but can override almost anything they want. `uhttp.NewClient()` is better than `uhttp.NewClient(defaultAuth, graph)`